django/core/management/commands/createcachetable.py

```diff
@@ -53,55 +54,33 @@ def create_table(self, database, tablename, dry_run):
                 self.stdout.write("Cache table '%s' already exists." % tablename)
             return
 
-        fields = (
+        class CacheTable(models.Model):
             # "key" is a reserved word in MySQL, so use "cache_key" instead.
-            models.CharField(name='cache_key', max_length=255, unique=True, primary_key=True),
-            models.TextField(name='value'),
-            models.DateTimeField(name='expires', db_index=True),
-        )
-        table_output = []
-        index_output = []
-        qn = connection.ops.quote_name
-        for f in fields:
-            field_output = [
-                qn(f.name),
-                f.db_type(connection=connection),
-                '%sNULL' % ('NOT ' if not f.null else ''),
-            ]
-            if f.primary_key:
-                field_output.append("PRIMARY KEY")
-            elif f.unique:
-                field_output.append("UNIQUE")
-            if f.db_index:
-                unique = "UNIQUE " if f.unique else ""
-                index_output.append(
-                    "CREATE %sINDEX %s ON %s (%s);" %
-                    (unique, qn('%s_%s' % (tablename, f.name)), qn(tablename), qn(f.name))
-                )
-            table_output.append(" ".join(field_output))
-        full_statement = ["CREATE TABLE %s (" % qn(tablename)]
-        for i, line in enumerate(table_output):
-            full_statement.append('    %s%s' % (line, ',' if i < len(table_output) - 1 else ''))
-        full_statement.append(');')
+            cache_key = models.CharField(max_length=255, primary_key=True)
+            value = models.TextField()
+            expires = models.DateTimeField(db_index=True)
 
-        full_statement = "\n".join(full_statement)
+            class Meta:
+                apps = Apps()
+                app_label = 'cache'
+                db_table = tablename
+                # constraints = [models.UniqueConstraint(fields=['cache_key'], name=tablename + '_cache_key')]
```

Comment:
I think this is unnecessary, was it `primary_key=True` that triggered the creation of `varchar_pattern_ops` index?

Reply:
Yes, the current code doesn't work because `primary_key=True` [implies `unique=True`](https://github.com/django/django/blob/53b6a466d89f972aa435fe7b5135e9b501ce68ee/django/db/models/fields/__init__.py#L731-L732) which triggers varchar_pattern_ops.

Maybe this is a justification for reopening https://code.djangoproject.com/ticket/24088 (text/varchar_pattern_ops on PostgreSQL could be optional)? I don't have a concrete proposal for a solution to that ticket though.

The other option I thought of was to collect the SQL, discard the like index if present, and then run SQL manually.

Reply:
> The other option I thought of was to collect the SQL, discard the like index if present, and then run SQL manually.

That could be an option. I guess we could also `editor.remove_index(Index(name="figure out right name", fields=['cache_key'], opsclasses=['varchar_pattern_ops'])` when `connection.vendor == 'postgresql'` but that seems brittle.

Fixing [#24082](https://code.djangoproject.com/ticket/24082) seems really challenging for now so I think a solution that circumvent it would be acceptable.